_drafts/conda.md

```diff
@@ -0,0 +1,716 @@
+README.md, or through a (sigh) guess-and-check method. If it builds, it works,
+right? (No)
+
+With Conda, the build and test loop becomes simple!
+
+```
+# Update all required packages
+$ conda env update -f environment.yml
+
+# Build and flash the firmware
+$ make && flash
+
+# Test the system
+
+# Mark as good or bad
+$ git bisect [good|bad]
+
+# Repeat many times...
+```
+
+### Distributing a Conda Environment
+
+If you want to be able to export and distribute a Conda environment, including
+Python, pip packages, and system packages, this is possible using the official
+Conda package called [Conda Pack](https://conda.github.io/conda-pack/).
+
+This is useful when you need to give a build or test environment to an end
+customer, a vendor, or a factory in China. It is also useful where the recipient
+has limited or no access to the Internet. These packaged environments can be
+used **without** installing Python, Conda, or any other packages on the host
+machine.
+
+## Creating Your Own Conda Packages
+
+There is a high chance that all packages required by your project, team, or
+developers are not yet built for Conda and hosted on the Conda Forge repository.
+Fret not, as building and publishing Conda packages is relatively simple. We'll
+quickly walk through how I built and published the `gcc-arm-none-eabi` package
+that is hosted [here](https://anaconda.org/memfault/gcc-arm-none-eabi) and built
+from the official
+[ARM GNU Embedded Toolchain](https://developer.arm.com/tools-and-software/open-source-software/developer-tools/gnu-toolchain/gnu-rm/downloads).
+
+Creating a package for this toolchain is straight-forward since the binaries are
+already built for each platform. All we need to do is download the tarball,
+extract it, and copy the contents into a Conda package.
+
+> If you want to jump ahead and look at the final recipe, it is hosted in
+> Memfault's
+> [conda-recipes Github repo](https://github.com/memfault/conda-recipes/tree/master/gcc-arm-none-eabi).
+
+### Creating the Build Environment
+
+The first thing we need to do is create an environment which has `conda-build`
+and `anaconda-client` installed, both of which will be used to build and publish
+our package.
+
+```
+# Create the Conda Environment we'll use for building
+$ conda create -n build
+
+# Activate the environment called `build`
+$ conda activate build
+
+# Install the necessary packages to build and publish Conda packages
+(build)
+$ conda install conda-build anaconda-client
+```
+
+Now we'll want to create an empty directory, with a single `meta.yaml` file
+within it.
+
+```
+$ mkdir /tmp/arm-none-eabi-gcc && cd /tmp/arm-none-eabi-gcc
+$ touch meta.yaml
+```
+
+### Defining the `meta.yaml` File
+
+Next, we'll set up our `meta.yaml` file with the following information, with
+each field described below. The definition is small because we are merely
+repackaging binaries.
+
+```yaml
+package:
+  name: gcc-arm-none-eabi
+  version: 8.2019.q3.update
```

Comment:
It might be too complicated for this post, but I like using jinja templating for the version, see the example here:
https://docs.conda.io/projects/conda-build/en/latest/resources/define-metadata.html

Reply:
Going to leave this out. Sadly, the Jinja confused me for the longest time, and I feel it might confuse others as well. I do like using it now that I get it :P